common/utils.js

```diff
@@ -1551,11 +1552,32 @@
             return getContextJSON().server ? getContextJSON().server.http : $http;
         };
 
+        /**
+         * Given the context, returns the value of the chaise config property for system column heuristics
+         * @params {String} context - the current app context
+         * @return {boolean|Array|null} boolean - value to show or hide all system columns
+         *                              Array - order of which columns to show
+         *                              null - no value defined for the current context
+         */
+        function systemColumnsMode(context) {
+            var cc = getConfigJSON();
+
+            var mode = null;
+            if (context == 'compact' && cc.SystemColumnsDisplayCompact)  {
```

Comment:
I think this should include all the sub-contexts of `compact` too. So the check should be `context.indexOf("compact") === 0`.

Reply:
I realized now that you've handled this in ermrestjs. But as I mentioned in the other PR, if you're sending a string called context, you should just send the whole context to chaise and let chaise deal with it. You don't need to do a pre-check in ermrestjs.